File: test/test-string.py
# hdf5-udf <file.h5> test-string.py Temperature.py:1000:double
def dynamic_dataset():
    input_data = lib.getData("Dataset1")
    input_dims = lib.getDims("Dataset1")
    udf_data = lib.getData("Temperature.py")
    udf_dims = lib.getDims("Temperature.py")

    for i in range(input_dims[0]):
        print(lib.string(input_data[i]), flush=True)

    for i in range(udf_dims[0]):
        udf_data[i] = i * 1.0

File: test/test-string.lua
-- hdf5-udf <file.h5> test-string.lua Temperature.lua:1000:double
function dynamic_dataset()
    local input_data = lib.getData("Dataset1")
    local input_dims = lib.getDims("Dataset1")
    local udf_data = lib.getData("Temperature.lua")
    local udf_dims = lib.getDims("Temperature.lua")

    for i=1, input_dims[1] do
        print(lib.string(input_data[i]))
    end

    for i=1, udf_dims[1] do
        udf_data[i] = (i-1) * 1.0
    end
end